src/Bicep.Core/Analyzers/Linter/ApiVersions/ApiVersionProviderFactory.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Runtime.CompilerServices;
+using Bicep.Core.Features;
+using Bicep.Core.Semantics.Namespaces;
+
+namespace Bicep.Core.Analyzers.Linter.ApiVersions;
+
+public class ApiVersionProviderFactory : IApiVersionProviderFactory
+{
+    private readonly ConditionalWeakTable<IFeatureProvider, IApiVersionProvider> apiVersionProviderCache = new();
```

Comment:
I'm going to remove this cache. The only place `IApiVersionProviderFactory::GetApiVersionProvider` is called is when a `SemanticModel` is instantiated inside a `Compilation` class, so it is effectively cached there, as users of the provider access it from the model's public `ApiVersionProvider` property.